src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -956,10 +977,20 @@ class OpenShiftDSL implements Serializable {
             if ( objectList != null ) {
                 return objectList;
             }
-
+            
             // Otherwise, we need to ask the API server what presently matches
-            OcAction.OcActionResult r = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("get", selectionArgs(), null, "-o=name") );
-            r.failIf( "Unable to retrieve object names: " + this.toString() );
+            OcAction.OcActionResult r = null;
+            ArrayList<String> g = new ArrayList<String>();
+            g.add("get")
+            OcAction.OcActionResult versionCheck = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("help", g, null, null) );
```

Comment:
Yeah I entertained that notion as well.  Ultimately though, I think we need to do this periodically instead of just onece since the user could import different versions of `oc` through Global Tool Configuration.

I then decided to keep it simple for now.  I should minimally have put a TOD comment referring to such an optimization.  Unless we have further debate here, I'll do that with the next push.